theano/sandbox/cuda/tests/test_cusolver.py

```diff
@@ -0,0 +1,71 @@
+from __future__ import absolute_import, division, print_function
+
+import unittest
+import numpy
+import theano
+
+from theano.tests import unittest_tools as utt
+
+# Skip tests if cuda_ndarray is not available.
+from nose.plugins.skip import SkipTest
+import theano.sandbox.cuda as cuda_ndarray
+from theano.misc.pycuda_init import pycuda_available
+from theano.sandbox.cuda.cusolver import cusolver_available
+
+from theano.sandbox.cuda import cusolver
+
+if not cuda_ndarray.cuda_available:
+    raise SkipTest('Optional package cuda not available')
+if not pycuda_available:
+    raise SkipTest('Optional package pycuda not available')
+if not cusolver_available:
+    raise SkipTest('Optional package scikits.cuda.cusolver not available')
+
+if theano.config.mode == 'FAST_COMPILE':
+    mode_with_gpu = theano.compile.mode.get_mode('FAST_RUN').including('gpu')
+else:
+    mode_with_gpu = theano.compile.mode.get_default_mode().including('gpu')
+
+
+class TestCusolver(unittest.TestCase):
+    def run_gpu_solve(self, A_val, x_val):
+        b_val = numpy.dot(A_val, x_val)
+        A = theano.tensor.matrix("A", dtype="float32")
+        b = theano.tensor.matrix("b", dtype="float32")
+
+        solver = cusolver.gpu_solve(A, b)
```

Comment:
Can you add a test for when trans is T instead of the default? Just to make sure it work.